src/main/java/edu/cornell/kfs/tax/batch/dataaccess/impl/TransactionDetailProcessorDaoJdbcImpl.java

```diff
@@ -0,0 +1,309 @@
+            try {
+                final TransactionDetailRowMapper<U> rowMapper = rowMapperFactory.createMapper(
+                        encryptionService, transactionDetailMetadata, resultSet);
+                return handler.performProcessing(config, rowMapper);
+            } catch (final IOException e) {
+                throw new UncheckedIOException(e);
+            } catch (final SQLException | RuntimeException e) {
+                throw e;
+            } catch (final Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    private CuSqlQuery createTransactionDetailQuery(
+            final TaxBatchConfig config, final TaxDtoDbMetadata transactionDetailMetadata) {
+        final TransactionDetailField taxBoxField = determineTaxBoxFieldToExamine(config);
+        final Criteria taxBoxCriteria;
+
+        if (config.getMode() == TaxBatchConfig.Mode.CREATE_TAX_FILES) {
+            taxBoxCriteria = Criteria.equal(taxBoxField, CUTaxConstants.NEEDS_UPDATING_BOX_KEY);
+        } else {
+            taxBoxCriteria = Criteria.notEqual(taxBoxField, CUTaxConstants.NEEDS_UPDATING_BOX_KEY);
+        }
+
+        return new TaxQueryBuilder(transactionDetailMetadata)
+                .selectAllMappedFields()
+                .from(TransactionDetail.class)
+                .where(
+                        Criteria.equal(TransactionDetailField.reportYear, Types.INTEGER, config.getReportYear()),
+                        taxBoxCriteria)
+                .orderBy(
+                        QuerySort.ascending(TransactionDetailField.vendorTaxNumber),
+                        QuerySort.ascending(TransactionDetailField.incomeCode),
+                        QuerySort.ascending(TransactionDetailField.incomeCodeSubType))
+                .build();
+    }
+
+    private TransactionDetailField determineTaxBoxFieldToExamine(final TaxBatchConfig config) {
+        switch (config.getTaxType()) {
+            case CUTaxConstants.TAX_TYPE_1099:
+                throw new RuntimeException("This implementation currently does not support 1099 processing");
+
+            case CUTaxConstants.TAX_TYPE_1042S:
+                return TransactionDetailField.form1042SBox;
+
+            default:
+                throw new IllegalStateException("Unrecognized tax type: " + config.getTaxType());
+        }
+    }
+
+
+
+    @Override
+    public VendorQueryResults getVendor(final Integer vendorHeaderId, final Integer vendorDetailId)
+            throws SQLException {
+        final Pair<Integer, Integer> vendorId = Pair.of(vendorHeaderId, vendorDetailId);
+        final TaxDtoDbMetadata vendorMetadata = taxTableMetadataLookupService
+                .getDatabaseMappingMetadataForDto(VendorField.class);
+        final CuSqlQuery query = createVendorQuery(vendorId, vendorMetadata);
+
+        return queryForResults(query,
+                resultSet -> extractVendorQueryResults(resultSet, vendorId, vendorMetadata));
```

Comment:
Is this a factory design pattern? Wouldn't it be simpiler to get the results on one line and then call the extract method on the next line?

Reply:
This lambda implements the JDBC-Template-related `ResultSetExtractor` interface, which is used by query operations where the callback code manually iterates over the `ResultSet` and returns an object representing the final results. (The `extractVendorQueryResults()` method performs this type of manual iteration over the `ResultSet`.) Using a `ResultSetExtractor` seemed like the best way to support queries that need an updatable `ResultSet`, and I decided to use such extractors for the DAO's other queries as well.